### ISSUE TITLE
AP = 0 when test scratch_e2e_faster_rcnn_R_50_FPN_Xconv1fc_3x_gn.yaml

### ISSUE THREAD

**Issue by dl19940602** [facebookresearch/maskrcnn-benchmark#1044]
Hi, i train coco with config scratch_e2e_faster_rcnn_R_50_FPN_Xconv1fc_3x_gn.yaml, it trains nomally and loss decrease to 0.3-0.4, when i test the model,the results of AP and AR are all 0.I use the syncbn instead of bn on backbone,and i use FPN without using GN. I don't what is the problem, could you tell me how can i do mow?

Follwing are my config and test log:
INPUT:
  MIN_SIZE_TRAIN: (800,)
  MAX_SIZE_TRAIN: 1333
  MIN_SIZE_TEST: 800
  MAX_SIZE_TEST: 1333
MODEL:
  META_ARCHITECTURE: "GeneralizedRCNN"
  WEIGHT: "" # no pretrained model
  BACKBONE:
    CONV_BODY: "S-20-FPN"
#    FREEZE_CONV_BODY_AT: 0 # finetune all layers
  RESNETS: # use GN for backbone
    BACKBONE_OUT_CHANNELS: 256
#    STRIDE_IN_1X1: False
#    TRANS_FUNC: "BottleneckWithGN"
#    STEM_FUNC: "StemWithGN"
  FPN:
    USE_GN: False # use GN for FPN
  RPN:
    USE_FPN: True
    ANCHOR_STRIDE: (4, 8, 16, 32, 64)
    PRE_NMS_TOP_N_TRAIN: 2000
    PRE_NMS_TOP_N_TEST: 1000
    POST_NMS_TOP_N_TEST: 1000
    FPN_POST_NMS_TOP_N_TEST: 1000
  ROI_HEADS:
    USE_FPN: True
    BATCH_SIZE_PER_IMAGE: 512
    POSITIVE_FRACTION: 0.25
  ROI_BOX_HEAD:
    USE_GN: True # use GN for bbox head
    POOLER_RESOLUTION: 7
    POOLER_SCALES: (0.25, 0.125, 0.0625, 0.03125)
    POOLER_SAMPLING_RATIO: 2
    CONV_HEAD_DIM: 256
    NUM_STACKED_CONVS: 4
    FEATURE_EXTRACTOR: "FPNXconv1fcFeatureExtractor"
    PREDICTOR: "FPNPredictor"
DATASETS:
  TRAIN: ("coco_2017_train",  )
  TEST: ("coco_2017_val",)
DATALOADER:
  SIZE_DIVISIBILITY: 32
SOLVER:
  # Assume 8 gpus
  BASE_LR: 0.02
  WEIGHT_DECAY: 0.0001
  STEPS: (210000, 250000)
  MAX_ITER: 270000
  IMS_PER_BATCH: 16
TEST:
  IMS_PER_BATCH: 8

test log:
Average Precision (AP) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.000
Average Precision (AP) @[ IoU=0.50 | area= all | maxDets=100 ] = 0.000
Average Precision (AP) @[ IoU=0.75 | area= all | maxDets=100 ] = 0.000
Average Precision (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] =0.000
Average Precision (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
Average Precision (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 1 ] = 0.000
Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 10 ] = 0.000
Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.000
Average Recall (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
Average Recall (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
Average Recall (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000

**Comment by liyz15** [comment]
Probably related to weight decay for SyncBN, see https://github.com/facebookresearch/maskrcnn-benchmark/issues/267#issuecomment-489516019

**Comment by zimenglan-sysu-512** [comment]
when u train the network, does the loss encounter `NaN`?

**Comment by dl19940602** [comment]
@zimenglan-sysu-512 the loss convergence normally,and it finally convergence to 0.3-0.4.

**Comment by rnsandeep** [comment]
Are you using the same coco dataset or using your own dataset? If you are using your own dataset check if the annotations boxes are correct.

**Comment by wykhan** [comment]
I have the same question.  I use "e2e_mask_rcnn_R_50_FPN_1x.yaml" and coco2017 datasets
